### ISSUE TITLE
Screen readers read out abbreviations as words

### ISSUE THREAD

**Issue by mnigh** [GCTC-NTGC/gc-digital-talent#3313]
### Description
Screen readers read out abbreviations as words abbreviations (AS-01, for example on Update Pool form is read out as the word _as_ instead of the letters _A_ _S_). 

### Challenges

- Difficult to control for any values from the database that can be added or modified outside of the codebase (a skill like IT Business Analyst / IT Project Management)
- Cannot add hidden text on something like a form select option

### Questions
Other than having the full version of the acronym (in the case of AS, it would be Administrative Services), do other ways exist to make abbreviations more accessible for screen readers?

Answer: not worried about reading full words out for classifications.

## proposed solution
Split up classifications - add a space between classification letters in an aria-label.

## Acceptance Criteria
- [ ] Classification is read out as "A" "S" instead of "as" by screen reader.

**Comment by patcon** [comment]
This is an interesting problem, as these will appear in a ton of option labels, and it feels odd to put html tags into database entries.

Couple thoughts:
- if we're saying "AS-01" (non-expanded) for sighted people is appropriate, then the issue for non-sighted ppl is better framed as "it should as ay-ess-one", rather than "it should say administrative services 1". If we think non-sighted people need the acronym expanded, then so do sighted people :)
- i understand that `<span aria-label="A.S. 1">AS-01</span>` is a way to force screen readers to pronounce properly. [ref](https://www.queensu.ca/accessibility/across-campus/web-standards-and-accessibility-development-guide/writing-web/acronyms-and)
- if we don't like the idea of storing html tags in database labels (which in theory are editable by admins who don't want to use html), then we can pattern match with regex to create html tags: https://regex101.com/r/eoPnrj/1
  - AS-01 => `<span aria-label="A.S. 1">AS-01</span>`
- we can process labels for this, and any other place we think these might show up.

**Comment by patcon** [comment]
> Cannot add hidden text on something like a form select option

I think this might be mitigated if/when we fully move to react-select, as their options and labels are full HTML

**Comment by MaddyDaigle** [comment]
So, I did some research, and all of these should be tagged as abbreviations` <abbr> ` with a title attribute defining the abbreviated text.  This allows the screen reader user to understand what the abbreviation means, even if it's not read out entirely correctly.

Here is a way you could code it so that it both has the abbreviation title and reads it out correctly.  I just tested this and it seems to work!

<abbr title="Administrative Services" aria-label="A S 1"><span aria-hidden="true">AS-01</span></abbr>

https://www.w3.org/TR/WCAG20-TECHS/H28.html

**Comment by MaddyDaigle** [comment]
@patcon could I assign this to you if you think the above solution will work? I'm not set up to make code changes right now. :(

**Comment by petertgiles** [comment]
Should consider month abbreviations, like on the experience accordions.

**Comment by patcon** [comment]
Thanks Maddy! I'm understanding that <attr title="foo"> will never be read, and is just semantic -- not related to accessibility, yes? Seems that CSS props are needed to actually affect how it's read

Just to toss another options out there, one of the recommendations at the bottom of [my linked Stack Overflow solution](https://stackoverflow.com/a/32892825) points to this:

[WCAG 2.0: G97: Providing the first use of an abbreviation immediately before or after the expanded form](https://www.w3.org/TR/2015/NOTE-WCAG20-TECHS-20150226/G97)

And that got me thinking that we could alternatively do this in the react-select dropdowns, and not use the abbr tag there:

https://codesandbox.io/s/codesandboxer-example-forked-p66zsi

<img width="549" alt="Screen Shot 2022-11-25 at 2 17 01 PM" src="https://user-images.githubusercontent.com/305339/204003596-d0ae255c-6374-4b3e-8ed4-be522a6d0cea.png">

**Comment by patcon** [comment]
Oh, and if you feel your part in research is done @MaddyDaigle, then feel free to unassign yourself, and someone will take this on once it's been added to a sprint :)

**Comment by gobyrne** [comment]
Eric & Matt think we can add an ARIA-Label, get the full name and put it in there.

**Comment by gobyrne** [comment]
Not actually doing full words for now. Just breaking up the classification letters.
